### PR TITLE
Fix quick open mochi

### PR DESCRIPTION
**Pull request by wldcordeiro**
Associated Issue: N/A

### Summary of Changes

* Fix up the quick open mochitest that had a failure. Not 100% certain if it was just racy before or if this was to adjust to changes from @jasonLaster's recent work. 🙁 
* Doc some tips based on what I did here.
* Prettier did its thing here.

### Test Plan

Example test plan:

- [x] `yarn mochi browser_dbg-quick-open.js`

### CODE REVIEW

File: docs/mochitests.md [hunk 1]
```diff
@@ -139,9 +139,17 @@ add_task(function* () {
 
 The Debugger Mochitest API Documentation can be found [here](https://devtools-html.github.io/debugger.html/reference#mochitest).
 
+## Debugging Tips
+
+* Run your mochitest with `yarn mochid <filename>` and add a `debugger` statement to your test before or after a failure.
+* Go back to the development server and watch the Redux logs (either with the Redux Devtools or by setting action logging on in your [local config][local-config].) as you perform your mochitest step by step. Watch the actions that fire. Sometimes all you need is a `waitForDispatch("THE_ACTION")` to ensure enough time has passed for your assertion to be true.
```
Comment (jasonLaster) [inline docs/mochitests.md]: that should now always be on for tests? maybe not though...

Comment (jasonLaster) [reply docs/mochitests.md]: `waitForDispatch(dbg, "THE_ACTION")`

File: src/test/mochitest/browser_dbg-quick-open.js [hunk 2]
```diff
@@ -52,64 +57,61 @@ function quickOpen(dbg, query, shortcut = "quickOpen") {
...
   let source = dbg.selectors.getSelectedSource(dbg.getState());
   ok(source.get("url").match(/switching-01/), "first source is selected");
 
-  info('Arrow keys and check to see if source is selected');
+  info("Arrow keys and check to see if source is selected");
   quickOpen(dbg, "sw");
   is(resultCount(dbg), 2);
   pressKey(dbg, "Down");
   pressKey(dbg, "Enter");
+  await waitForDispatch(dbg, "SELECT_SOURCE");
```
Comment (jasonLaster) [inline src/test/mochitest/browser_dbg-quick-open.js]: maybe we can make `waitForSelectedSource` smarter... that will help make the tests easier to write

```diff
diff --git a/src/test/mochitest/head.js b/src/test/mochitest/head.js
index 51d659f..e57ff70 100644
--- a/src/test/mochitest/head.js
+++ b/src/test/mochitest/head.js
@@ -218,16 +218,26 @@ async function waitForElement(dbg, selector) {
   return findElementWithSelector(dbg, selector);
 }
 
-function waitForSelectedSource(dbg, sourceId) {
+function waitForSelectedSource(dbg, url) {
   return waitForState(dbg, state => {
     const source = dbg.selectors.getSelectedSource(state);
-    const isLoaded =
-      source && source.has("loadedState") && sourceUtils.isLoaded(source);
-    if (sourceId) {
-      return isLoaded && sourceId == source.get("id");
+    const isLoaded = source && sourceUtils.isLoaded(source);
+    if (!isLoaded) {
+      return false;
     }
 
-    return isLoaded;
+    if (!url) {
+      return true;
+    }
+
+    const newSource = findSource(dbg, url)
+    if (newSource.id != source.get("id")) {
+      return false;
+    }
+
+    const symbols = dbg.selectors.getSymbolsForSource(state, source.id)
+    return symbols;
+
   }, "selected source");
 }
 
```

File: docs/mochitests.md [hunk 3]
```diff
@@ -139,9 +139,17 @@ add_task(function* () {
 
 The Debugger Mochitest API Documentation can be found [here](https://devtools-html.github.io/debugger.html/reference#mochitest).
 
+## Debugging Tips
+
+* Run your mochitest with `yarn mochid <filename>` and add a `debugger` statement to your test before or after a failure.
```
Comment (jasonLaster) [inline docs/mochitests.md]: screenshots of the test paused in the debugger code and the test code would help show what it is happening

File: src/test/mochitest/head.js [hunk 4]
```diff
@@ -218,18 +218,27 @@ async function waitForElement(dbg, selector) {
...
+        return true;
+      }
+
+      const newSource = findSource(dbg, url);
+      if (newSource.id != source.get("id")) {
+        return false;
+      }
+
+      // wait for async work to be done
+      return dbg.selectors.hasSymbols(state, source.toJS());
```
Comment (wldcordeiro) [inline src/test/mochitest/head.js]: @jasonLaster you didn't have the `toJS` call here which is what was making us hang.

**Pull request merged**